torch/testing/_internal/distributed/nn/api/remote_module_test.py

```diff
@@ -385,6 +391,31 @@ def hook(module, grad_input, grad_output):
             ):
                 remote_module.extra_repr()
 
+    @dist_utils.dist_init
```

Comment:
We should add a test where we validate that if a user adds a new attribute to RemoteModule and it is not in `_REMOTE_MODULE_PICKLED_ATTRIBUTES` we fail. It could be possible that the new attribute might not need to be pickled.

In that case, we could have another list `_REMOTE_MODULE_ATTRIBUTES_IGNORE_FOR_PICKLING`. In this case, our unit test should validate that each attribute of RemoteModule is either in `_REMOTE_MODULE_PICKLED_ATTRIBUTES` or `_REMOTE_MODULE_ATTRIBUTES_IGNORE_FOR_PICKLING`. This way it forces developers to add a new attribute to either one of those lists.

Reply:
Added _REMOTE_MODULE_ATTRIBUTES_IGNORE_FOR_PICKLING as well as a unit test.